pennylane_lightning/core/src/simulators/lightning_gpu/utils/cuda_helpers.hpp

```diff
@@ -0,0 +1,391 @@
+ * PennyLane-Lightning-GPU device. Minimum supported architecture is SM 7.0.
+ *
+ * @param device_number GPU device index
+ * @return bool
+ */
+static bool isCuQuantumSupported(int device_number = 0) {
+    cudaDeviceProp deviceProp;
+    cudaGetDeviceProperties(&deviceProp, device_number);
+    return deviceProp.major >= 7;
+}
+
+/**
+ * @brief Get current GPU major.minor support
+ *
+ * @param device_number
+ * @return std::pair<int,int>
+ */
+static std::pair<int, int> getGPUArch(int device_number = 0) {
+    cudaDeviceProp deviceProp;
+    cudaGetDeviceProperties(&deviceProp, device_number);
+    return std::make_pair(deviceProp.major, deviceProp.minor);
+}
+
+inline static auto pauliStringToEnum(const std::string &pauli_word)
+    -> std::vector<custatevecPauli_t> {
+    // Map string rep to Pauli enums
+    const std::unordered_map<std::string, custatevecPauli_t> pauli_map{
+        std::pair<const std::string, custatevecPauli_t>{std::string("X"),
+                                                        CUSTATEVEC_PAULI_X},
+        std::pair<const std::string, custatevecPauli_t>{std::string("Y"),
+                                                        CUSTATEVEC_PAULI_Y},
+        std::pair<const std::string, custatevecPauli_t>{std::string("Z"),
+                                                        CUSTATEVEC_PAULI_Z},
+        std::pair<const std::string, custatevecPauli_t>{std::string("I"),
+                                                        CUSTATEVEC_PAULI_I}};
+
+    static constexpr std::size_t num_char = 1;
+
+    std::vector<custatevecPauli_t> output;
+    output.reserve(pauli_word.size());
+
+    for (const auto ch : pauli_word) {
+        auto out = pauli_map.at(std::string(num_char, ch));
+        output.push_back(out);
+    }
+    return output;
+}
+
+inline static auto pauliStringToOpNames(const std::string &pauli_word)
+    -> std::vector<std::string> {
+    // Map string rep to Pauli
+    const std::unordered_map<std::string, std::string> pauli_map{
+        std::pair<const std::string, std::string>{std::string("X"),
+                                                  std::string("PauliX")},
+        std::pair<const std::string, std::string>{std::string("Y"),
+                                                  std::string("PauliY")},
+        std::pair<const std::string, std::string>{std::string("Z"),
+                                                  std::string("PauliZ")},
+        std::pair<const std::string, std::string>{std::string("I"),
+                                                  std::string("Identity")}};
+
+    static constexpr std::size_t num_char = 1;
+
+    std::vector<std::string> output;
+    output.reserve(pauli_word.size());
+
+    for (const auto ch : pauli_word) {
+        auto out = pauli_map.at(std::string(num_char, ch));
+        output.push_back(out);
+    }
+    return output;
+}
+
+/**
+ * Utility hash function for complex vectors representing matrices.
+ */
+struct MatrixHasher {
```

Comment:
Appears unused, consider removing.

Reply:
This should be required for the gate-caching functionality. We need a unique ID for the gate matrix being cached. I'd be surprised if this isn't used by the binding layer.

Reply:
`MatrixHasher` was defined in PR#62[https://github.com/PennyLaneAI/pennylane-lightning-gpu/pull/62] and used as private data of `HermitianObsGPU`. `gate_id_hash` is defined here in the `GateCache` class. Let me remove it.

Reply:
As mentioned, it was used for the unique string naming output for  Hermitian observables. Let's keep this in and make sure it is used in the same location as before.